apps/studio/components/interfaces/Storage/StorageMenuV2.tsx

```diff
@@ -1,29 +1,47 @@
-import { useParams } from 'common'
 import Link from 'next/link'
+import { useRouter } from 'next/router'
+
+import { IS_PLATFORM, useParams } from 'common'
 import { Menu } from 'ui'
 import { BUCKET_TYPES, BUCKET_TYPE_KEYS, DEFAULT_BUCKET_TYPE } from './Storage.constants'
 
 export const StorageMenuV2 = () => {
-  const { ref, bucketType } = useParams()
-  const selectedBucketType = bucketType || DEFAULT_BUCKET_TYPE
+  const router = useRouter()
+  const { ref } = useParams()
+
+  const page = router.pathname.split('/')[4] as undefined | 'files' | 'analytics' | 'vectors' | 's3'
```

Comment:
Now includes root S3 item. See other comment about it now being a shared setting across storage types ([context](https://supabase.slack.com/archives/C0161K73J1J/p1759385963488829?thread_ts=1759303258.525489&cid=C0161K73J1J)).